src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacItemQuestionValidator.java

```diff
@@ -116,12 +123,22 @@ public final QuestionValidationResponse validateQuestionResponse(final Question
                 }
 
                 // ... look for a match to the submitted answer:
-                if (itemChoice.getItems().size() != submittedChoice.getItems().size()) {
-                    continue;
-                }
+
                 Set<String> choiceItemIds = itemChoice.getItems().stream().map(Item::getId).collect(Collectors.toSet());
 
-                if (choiceItemIds.equals(submittedItemIds)) {
+                // Do not allow subset matching by default
+                boolean allowSubsetMatch = (null != itemChoice.isAllowSubsetMatch() && itemChoice.isAllowSubsetMatch());
+
+                /* If the intersection of the submitted and choice ids is equal to the submitted ones, then
+                   this means that:
+                    - submittedItemIds.size() <= choiceItemIds.size()
+                    - All submitted ids are within the set of choice ids
+                 */
+                if (allowSubsetMatch && Sets.intersection(submittedItemIds, choiceItemIds).equals(submittedItemIds)) {
```

Comment:
This feels the wrong way round to me?

If the question has options {A, B, C, D} and the correct answer is {A, B, C} and a known wrong answer that allows subset matches is just {D}, then a student answer of {A, B, C, D} should match this known wrong answer since that's what we need to allow to avoid writing every possible combination that includes D if we just want to say "D is definitely wrong".

I think this should be `Sets.intersection(submittedItemIds, choiceItemIds).equals(choiceItemIds)`, i.e. are all of the items from this choice in the submission?

Reply:
What you have here (and is demonstrated in your `isaacItemQuestionValidator_SubsetOfCorrectItems_CorrectResponseShouldBeReturned` test) is exactly the opposite of what this is meant to do. If we wanted to allow any subset of the options {A, B, C} to be marked as correct, then in "my" version you'd just add {A}, {B}, and {C} as correct choices that allow subset matches and that would work too. But that's not a case that I can see happening.